src/software/replay/replay_test.cpp

```diff
@@ -0,0 +1,88 @@
+#include <google/protobuf/util/message_differencer.h>
+#include <gtest/gtest.h>
+
+#include "software/multithreading/subject.h"
+#include "software/replay/replay_logger.h"
+#include "software/replay/replay_reader.h"
+
+constexpr const char* REPLAY_TEST_PATH_SUFFIX = "software/replay/test_replay";
+
+namespace fs = std::experimental::filesystem;
+
+// hahaha test subject because it's a subject but also a test subject
+class TestSubject : public Subject<SensorMsg>
+{
+   public:
+    void sendValue(SensorMsg val)
+    {
+        sendValueToObservers(val);
+    }
+};
+
+TEST(ReplayTest, test_read_and_write_replay)
+{
+    // unfortunately due to the unavailablity of ordering tests and functions that
+    // generate test files that actually work, we have to read recorded replays back,
+    // write them, and then read them again in the same test
+    // TODO: record replay data with refbox running
```

Comment:
TODO should be linked to a ticket if not already done

Reply:
#1584